x-pack/plugins/cases/public/components/create/flyout/use_cases_add_to_new_case_flyout.tsx

```diff
@@ -27,39 +28,43 @@ export const useCasesAddToNewCaseFlyout = (props: AddToNewCaseFlyoutProps) => {
     });
   }, [dispatch]);
 
-  const openFlyout = useCallback(() => {
-    dispatch({
-      type: CasesContextStoreActionsList.OPEN_CREATE_CASE_FLYOUT,
-      payload: {
-        ...props,
-        onClose: () => {
-          closeFlyout();
-          if (props.onClose) {
-            return props.onClose();
-          }
-        },
-        onSuccess: async (theCase: Case) => {
-          if (theCase) {
-            casesToasts.showSuccessAttach({
-              theCase,
-              attachments: props.attachments,
-              title: props.toastTitle,
-              content: props.toastContent,
-            });
-          }
-          if (props.onSuccess) {
-            return props.onSuccess(theCase);
-          }
+  const openFlyout = useCallback(
+    (attachments?: CaseAttachments) => {
+      dispatch({
+        type: CasesContextStoreActionsList.OPEN_CREATE_CASE_FLYOUT,
+        payload: {
+          ...props,
+          attachments: attachments ?? props.attachments,
```

Comment:
I am confused, what is the difference between `attachments` and `prop.attachments`?

Reply:
attachments is a parameter to this function and `props.attachments` are the props coming when you first call the hook.

Reply:
It is effectively a way to overwrite the originally passed attachments.

Reply:
sure, but why do we want to override them? why are we receiving the attachments in the hook props, and in the callback as well? My gut feeling says one of them is not needed, unless there's some special case to justify it.

Reply:
this hooks was initially created to accept the attachments before you open the modal/flyout, but in this case when you are adding a bulk, you can't really know which attachments are going to be send when you invoke the hook, only on the moment you open the modal/flyout. Therefore I implemented this overwrite.